### ISSUE TITLE
Static UI: Bounds lock not firm

### ISSUE THREAD

**Issue by amyleew** [mapbox/mapbox-studio-classic#500]
@camilleanne in prepping this style, i noticed that the bounding lock button is not actually locking the proportions accurately.

when i _lock_ a dimension, then drag the top left corner to move it over my area on the map... it slightly resizes the locked dimensions. 

**gif walkthru:**
![print-bounding-lock](https://cloud.githubusercontent.com/assets/4587826/3447053/204e1d4e-0149-11e4-9984-ef50e35cb0e5.gif)

/cc @lxbarth @nickidlugash

**Comment by camilleanne** [comment]
@amyleew isn't web-mercator awesome? It's getting stretchy near the poles-- if you notice, the resize handles on the left and right sides aren't exactly centered anymore. That's because of [this magic](http://en.wikipedia.org/wiki/Mercator_projection#mediaviewer/File:Mercator_projection_SW.jpg) that makes the boxes near the equator square the ones near the poles all rectangular and long. Notice that the x (width) dimension won't be super off (there is a margin of error related related to rounding, however), but the y (height) dimensions will be.

Everything needs to transition from geo bounds -> pixel bounds. I'll take a look tonight/tomorrow.

**Comment by amyleew** [comment]
> isn't web-mercator awesome?

@camilleanne and sneaky too... thanks for once again working your magic!

**Comment by camilleanne** [comment]
@amyleew finally had some time to take a look at this.

I've got a hacky first draft in the mercator-adjust branch of tm2 (https://github.com/mapbox/tm2/compare/mercator-adjust), but there are some dragging issues related to recalculating to maintain an unskewed rectangle. I'll continue investigating, but there may need to be a lesser of two evils choice.

**Comment by amyleew** [comment]
@camilleanne - pulled mercator-adjust but can't really tell the difference? where should i be testing?

**Comment by camilleanne** [comment]
So you're only going to see those issues from the gif above closer to the poles where geographic space has been skewed to fit into a rectangular space -- and you'll only see it going north to south, or south to north, the west/east dimensions are unaffected. The fix I made only kicks in if the dimensions are locked. What I noticed last night is when you place the bounding box up near one of the poles (e.g. northern Canada) and then start panning around, for some reason the bounding box moves a bit as well. Still figuring that out. But it is producing images that are consistent with the input dimensions.

![](http://upload.wikimedia.org/wikipedia/commons/thumb/f/f4/Mercator_projection_SW.jpg/1920px-Mercator_projection_SW.jpg)
